exchanges/request/request.go

```diff
@@ -122,27 +133,46 @@ func (r *Requester) doRequest(req *http.Request, p *Item) error {
 		}
 	}
 
-	var timeoutError error
-	for i := 0; i < r.timeoutRetryAttempts+1; i++ {
+	for attempt := 1; ; attempt++ {
 		// Initiate a rate limit reservation and sleep on requested endpoint
 		err := r.InitiateRateLimit(p.Endpoint)
 		if err != nil {
 			return err
 		}
 
 		resp, err := r.HTTPClient.Do(req)
-		if err != nil {
-			if timeoutErr, ok := err.(net.Error); ok && timeoutErr.Timeout() {
-				if p.Verbose {
-					log.Errorf(log.RequestSys,
-						"%s request has timed-out retrying request, count %d",
-						r.Name,
-						i)
+		if retry, checkErr := r.retryPolicy(resp, err); checkErr != nil {
+			return checkErr
+		} else if retry {
+			if err == nil {
+				// If the body isn't fully read, the connection cannot be re-used
+				r.drainBody(resp.Body)
+			}
+
+			if attempt > r.maxRetries {
+				if err != nil {
+					return fmt.Errorf("request.go error - failed to retry request, err: %v", err)
 				}
-				timeoutError = err
-				continue
+				return fmt.Errorf("request.go error - failed to retry request, status: %s", resp.Status)
```

Comment:
Example error from here:

```
[ERROR] | 23/04/2020 22:14:36 | Binance failed to update tradable pairs. Err: request.go error - failed to retry request, status: 418 I'm a teapot
```

i.e. Binance response status for continuing to exceed the rate limit.